src/main/java/edu/suffolk/litlab/efspserver/services/CodesService.java

```diff
@@ -0,0 +1,280 @@
+package edu.suffolk.litlab.efspserver.services;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.suffolk.litlab.efspserver.codes.CaseCategory;
+import edu.suffolk.litlab.efspserver.codes.CaseType;
+import edu.suffolk.litlab.efspserver.codes.CodeDatabase;
+import edu.suffolk.litlab.efspserver.codes.DataFieldRow;
+import edu.suffolk.litlab.efspserver.codes.Disclaimer;
+import edu.suffolk.litlab.efspserver.codes.DocumentTypeTableRow;
+import edu.suffolk.litlab.efspserver.codes.FileType;
+import edu.suffolk.litlab.efspserver.codes.FilingCode;
+import edu.suffolk.litlab.efspserver.codes.FilingComponent;
+import edu.suffolk.litlab.efspserver.codes.NameAndCode;
+import edu.suffolk.litlab.efspserver.codes.PartyType;
+
+@Path("/codes/")
+@Produces({MediaType.APPLICATION_JSON})
+public class CodesService {
+  private static Logger log = LoggerFactory.getLogger(CodesService.class); 
+  private Map<String, EfmFilingInterface> filingInterfaces;
+
+  private CodeDatabase cd;
+  public CodesService(CodeDatabase cd, Map<String, EfmFilingInterface> filingInterfaces) {
+    this.cd = cd;
+    this.filingInterfaces = filingInterfaces;
+  }
+
+  @GET
+  @Path("/courts/")
```

Comment:
Nit, but I've been going by the rule of thumb that endpoints are supposed to not have the trailing slash.